### ISSUE TITLE
Error getting JDBC meta data

### ISSUE THREAD

**Issue by markelhas** [yugabyte/yugabyte-db#13110]
Jira Link: [DB-2818](https://yugabyte.atlassian.net/browse/DB-2818)
### Description

I've just upgrade my cluster to latest version of yugabyteDB, v2.12.7.0 - June 24, 2022, Build: 2.12.7.0-b27, and after the deployment I'm not able to connect with DBeaver anymore!
I'm able to exec to my pod and use ysql with success, everything seams working fine, but not able to connect using DBeaver.

```
Error getting JDBC meta data
  This connection has been closed.
org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:877)
	at org.postgresql.jdbc.PgConnection.getMetaData(PgConnection.java:1309)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCConnectionImpl.getMetaData(JDBCConnectionImpl.java:334)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initialize(JDBCDataSource.java:430)
	at org.jkiss.dbeaver.ext.postgresql.model.PostgreDataSource.initialize(PostgreDataSource.java:387)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.openDataSource(DataSourceDescriptor.java:990)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:920)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.runSync(ConnectJob.java:98)
	at org.jkiss.dbeaver.ui.actions.datasource.DataSourceHandler.connectToDataSource(DataSourceHandler.java:113)
	at org.jkiss.dbeaver.ui.actions.datasource.UIServiceConnectionsImpl.initConnection(UIServiceConnectionsImpl.java:70)
	at org.jkiss.dbeaver.model.navigator.DBNDataSource.initializeNode(DBNDataSource.java:158)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:225)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:80)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:88)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:72)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by ddorian** [comment]
I could reproduce on 2.12.7.0-b0. 
It works on 2.15.0.0.
A core dump gets created.
But I couldn't get the backtrace because it was giving " is not a core dump: File format not recognized".
[the_core_dump.zip](https://github.com/yugabyte/yugabyte-db/files/9018894/the_core_dump.zip)

**Comment by HarshDaryani896** [comment]
This issue got fixed and was backported to [2.12.9](https://phorge.dev.yugabyte.com/rYBDB7cc503b25ae69202ebfe09b8019e6b4cec9faff8)

Have verified connection from Dbeaver 23.2.0 with the following versions of the database : 
2.12.12.0, 2.14.12.0, 2.16.6.1, 2.18.2.1 and 2.19.0.0
